ilc/client/asyncBootUp.spec.js

```diff
@@ -0,0 +1,147 @@
+            spaBundle: 'here/footer-spa-bundle.js',
+            cssBundle: 'here/footer-css-bundle.css',
+            dependencies: {
+                footerDependency: 'here/footer-dependency',
+                footerThirdPartyDependency: 'there/footer-third-party-dependency',
+            },
+        }
+    };
+
+    const slots = {
+        ready: {
+            id: 'ready',
+        },
+        navbar: {
+            id: 'navbar',
+        },
+        body: {
+            id: 'body',
+        },
+        footer: {
+            id: 'footer',
+        },
+        resetRef: () => {
+            slots.ref = html`
+                <main>
+                    <script>window.ilcApps = ['${slots.ready.id}'];</script>
+                    <div>
+                        <div id=${slots.ready.id}">
+                            <div>
+                                Hello! I am Ready SPA.
+                                I should be marked as ready at once.
+                            </div>
+                        </div>
+                        <div id="${slots.navbar.id}">
+                            <div>
+                                Hello! I am Navbar SPA.
+                                I don't have SPA override config, so I should not override initial SPA config.
+                            </div>
+                        </div>
+                        <div id="${slots.body.id}">
+                            <div>
+                                Hello! I am Body SPA.
+                                I have SPA override config, so I should override initial SPA config.
+                            </div>
+                            <script type="spa-config-override">${JSON.stringify(config.body)}</script>
+                        </div>
+                        <div id="${slots.footer.id}">
+                            <div>
+                                Hello! I am Footer SPA.
+                                I have SPA override config, but I am going to be ready after single\`spa routing event.
+                                It means I don't have to mark myself as ready and I should not override initial SPA config.
+                            </div>
+                            <script type="spa-config-override">${JSON.stringify(config.footer)}</script>
+                        </div>
+                    </div>
+                </main>
+            `;
+        },
+    };
+
+
+    const overrideImportMap = sinon.spy(window.System, 'overrideImportMap');
+
+    let clock;
+
+    beforeEach(() => {
+        slots.resetRef();
+        document.body.appendChild(slots.ref);
+        clock = sinon.useFakeTimers();
+    });
+
+    afterEach(() => {
+        clock.restore();
+        document.body.removeChild(slots.ref);
+        overrideImportMap.resetHistory();
+    });
+
+    it('should wait for slots before single-spa`s routing-event, mark apps as ready if they exist and return override configs then', async () => {
```

Comment:
1. I don't think that it's acceptable to have a single test to cover all the logic within `asyncBootup.js`. This particular test is too generic and it's hard to understand it.
2. You actually haven't tested quite a lot of the use cases 
![image](https://user-images.githubusercontent.com/1125900/81281102-157e1280-9062-11ea-96ab-5ee199ce0457.png)